src/notebooks/debugger/kernelDebugAdapter.ts

```diff
@@ -22,7 +22,18 @@ export class KernelDebugAdapter extends KernelDebugAdapterBase {
             const response = await this.session.customRequest('dumpCell', {
                 code: cell.document.getText().replace(/\r\n/g, '\n')
             });
-            const norm = path.normalize((response as IDumpCellResponse).sourcePath);
+
+            // Based on the response, check if the kernel runs on a Unix or Windows machine.
+            // If it runs on Windows, then path normalization might be necessary
+            // (see https://github.com/ipython/ipykernel/issues/995).
+            const dumpFilePathOnKernelSide = (response as IDumpCellResponse).sourcePath;
+            var norm = '';
+            if (dumpFilePathOnKernelSide.match('[A-Za-z]:.*')) {
+                norm = path.normalize(dumpFilePathOnKernelSide);
```

Comment:
normalize would have the same cross-OS problem here. If you know it's a windows path then you could do `path.win32.normalize` or write a simpler function for the slashes